coding-articles/Article.md

```diff
@@ -0,0 +1,584 @@
+    name: "Detects all deposit transactions",
+    description,
+    alertId: `deposit-${alertId}`,
+    severity,
+    type,
+    protocol: "Depositor",
+    metadata: {
+      account,
+      depositedAmount,
+    },
+  });
+};
+```
+
+The `utils` the file is ready. You can check it [here](./code/src/utils.ts); the last step in building the bot is the implementation of `handleTransaction` function
+go to `src/agent.ts`
+
+```
+// below provideIntialize function
+// should return a function that will be used by the Bot
+export const provideHandleTransaction = (): HandleTransaction => async (txEvent: TransactionEvent) => {
+ // write code here
+}
+export default {
+ initialize: provideInitialize(utils.provider),
+ handleTransaction: provideHandleTransaction(networkManager)
+};
+```
+after each tx emitted on the blockchain Forta bot return an array of findings
+
+```
+async (txEvent: TransactionEvent) => {
+    const findings: Finding[] = [];
+}
+```
+
+Use filter log to get all `Deposited` event trigger on your smart contract
+
+```
+// below the finding array
+const depositLogs = txEvent.filterLog(
+      utils.EVENT_ABI,
+      networkManager.get("address") 
+    );
+```
+
+The `networkManager.get("address")` will return the smart contract address placed [here](./code/src/network.ts#L9).
+
+The last step is to start emitting the finding.
+
+```
+depositLogs.forEach((log: LogDescription) => {
+      const amount = ethers.utils.formatEther(log.args.amount);
+      // Emit finding for new deposit
+      findings.push(
+        utils.createFinding({
+          alertId: "001",
+          account: txEvent.transaction.from,
+          depositedAmount: amount,
+          description: "New deposit!",
+          severity: FindingSeverity.Low,
+          type: FindingType.Info,
+        })
+      );
+      // Emit finding if deposit less than the minimum amount
+      if (Number(amount) < networkManager.get("minimumDepositAmount")) {
+        findings.push(
+          utils.createFinding({
+            alertId: "002",
+            account: txEvent.transaction.from,
+            depositedAmount: amount,
+            description: "Someone deposited a very small amount",
+            severity: FindingSeverity.Medium,
+            type: FindingType.Suspicious,
+          })
+        );
+      }
+      // Emit finding for blacklisted addresses
+      if (
+        networkManager
+          .get("blacklistedAddresses")
+          .includes(txEvent.transaction.from)
+      ) {
+        findings.push(
+          utils.createFinding({
+            alertId: "003",
+            account: txEvent.transaction.from,
+            depositedAmount: amount,
+            description: "Blacklisted addresses deposit!",
+            severity: FindingSeverity.Critical,
+            type: FindingType.Exploit,
+          })
+        );
+      }
+    });
+return findings;
+```
+
+And that's it. Now your [bot](./code/src/agent.ts) is ready to use, just run
+```
+npm start
```

Comment:
The bot does work for me after a bit oof tinkering. I got two alers for a single txn I sent to your contract from this address.
_0x439356ad40d2f2961c99ffed4453f482aec453af_

This is one of the most technically intense articles I have gone through. Kudos on being able to work with such a cool piece of technology. Fortabot is awesome.
I spent my last 3 days looking into your article and fortabot in general, I still don't think I understand everything.

![image](https://user-images.githubusercontent.com/56264430/195696744-13eafd7d-b73e-49c4-82a3-058f8a20a566.png)

Reply:
The third alert `Blacklisted addresses deposit!` will work only if you add this address "0x439356ad40d2f2961c99ffed4453f482aec453af" inside the blacklist. in `src/networks.ts`

Reply:
The third alert is `Blacklisted addresses deposit!` to trigger this alert, you need to add this address 0x439356ad40d2f2961c99ffed4453f482aec453af on [src/networks.ts](https://github.com/yehia67/developer-hub-content/blob/master/coding-articles/code/src/network.ts#L11)